File: Encrypt and decrypt connections string in a projects configuration file/[C#][VB]-Encrypt and decrypt connections string in a projects configuration file/C# and VB/ConfigurationLibrary_cs/ConnectionProtection.cs
using System;
using System.Configuration;
using System.IO;

namespace ConfigurationLibrary_cs
{
    public class ConnectionProtection
    {
        public string FileName { get; set; }
        public ConnectionProtection(string ExecutableFileName)
        {
            if (!(File.Exists(string.Concat(ExecutableFileName, ".config"))))
            {
                throw new FileNotFoundException(string.Concat(ExecutableFileName, ".config"));
            }
            FileName = ExecutableFileName;
        }
        private bool EncryptConnectionString(bool encrypt, string fileName)
        {
            bool success = true;
            Configuration configuration = null;

            try
            {

                configuration = ConfigurationManager.OpenExeConfiguration(fileName);
                ConnectionStringsSection configSection = configuration.GetSection("connectionStrings") 
                    as ConnectionStringsSection;

                if ((!(configSection.ElementInformation.IsLocked)) && (!(configSection.SectionInformation.IsLocked)))
                {
                    if (encrypt && (!configSection.SectionInformation.IsProtected))
                    {
                        // encrypt the file
                        configSection.SectionInformation.ProtectSection("DataProtectionConfigurationProvider");
                    }

                    if ((!encrypt) && configSection.SectionInformation.IsProtected) //encrypt is true so encrypt
                    {
                        // decrypt the file. 
                        configSection.SectionInformation.UnprotectSection();
                    }

                    configSection.SectionInformation.ForceSave = true;
                    configuration.Save();

                    success = true;

                }
            }
            catch (Exception)
            {
                success = false;
            }

            return success;

        }
        public bool IsProtected()
        {
            Configuration configuration = ConfigurationManager.OpenExeConfiguration(FileName);
            ConnectionStringsSection configSection = configuration.GetSection("connectionStrings") as ConnectionStringsSection;
            return configSection.SectionInformation.IsProtected;
        }
        public bool EncryptFile()
        {
            if (File.Exists(FileName))
            {
                return EncryptConnectionString(true, FileName);
            }
            else
            {
                return false;
            }
        }
        public bool DecryptFile()
        {
            if (File.Exists(FileName))
            {

                return EncryptConnectionString(false, FileName);

            }
            else
            {
                return false;
            }
        }

    }
}

File: Encrypt and decrypt connections string in a projects configuration file/[C#][VB]-Encrypt and decrypt connections string in a projects configuration file/C# and VB/ConfigurationLibrary_vb/ConnectionProtection.vb
Imports System.Configuration
Imports System.IO
Public Class ConnectionProtection
    Public Property FileName As String
    Public Sub New(ByVal ExecutableFileName As String)
        If Not File.Exists(String.Concat(ExecutableFileName, ".config")) Then
            Throw New FileNotFoundException(String.Concat(ExecutableFileName, ".config"))
        End If
        FileName = ExecutableFileName
    End Sub
    Private Function EncryptConnectionString(ByVal encrypt As Boolean, ByVal fileName As String) As Boolean
        Dim success As Boolean = True
        Dim configuration As Configuration = Nothing

        Try

            configuration = ConfigurationManager.OpenExeConfiguration(fileName)
            Dim configSection As ConnectionStringsSection =
                TryCast(configuration.GetSection("connectionStrings"), ConnectionStringsSection)

            If (Not (configSection.ElementInformation.IsLocked)) AndAlso (Not (configSection.SectionInformation.IsLocked)) Then
                If encrypt AndAlso (Not configSection.SectionInformation.IsProtected) Then
                    ' encrypt the file
                    configSection.SectionInformation.ProtectSection("DataProtectionConfigurationProvider")
                End If

                If (Not encrypt) AndAlso configSection.SectionInformation.IsProtected Then 'encrypt is true so encrypt
                    ' decrypt the file. 
                    configSection.SectionInformation.UnprotectSection()
                End If

                configSection.SectionInformation.ForceSave = True
                configuration.Save()

                success = True

            End If
        Catch ex As Exception
            success = False
        End Try

        Return success

    End Function
    Public Function IsProtected() As Boolean
        Dim configuration As Configuration = ConfigurationManager.OpenExeConfiguration(FileName)
        Dim configSection As ConnectionStringsSection =
            TryCast(configuration.GetSection("connectionStrings"), ConnectionStringsSection)
        Return configSection.SectionInformation.IsProtected
    End Function
    Public Function EncryptFile() As Boolean
        If File.Exists(FileName) Then
            Return EncryptConnectionString(True, FileName)
        Else
            Return False
        End If
    End Function
    Public Function DecryptFile() As Boolean
        If File.Exists(FileName) Then

            Return EncryptConnectionString(False, FileName)

        Else
            Return False
        End If
    End Function

End Class
